operator/v1/types_ingress.go

```diff
@@ -282,6 +287,339 @@ var (
+	//     - !EDH-DSS-DES-CBC3-SHA
+	//     - !KRB5-DES-CBC3-SHA
+	//     - !SRP
+	//   securityProtocol:
+	//     minimumVersion: TLSv1.0
+	//     maximumVersion: TLSv1.2
+	//   dhParamSize: 1024
+	//
+	// +optional
+	// +nullable
+	Old *OldSecurityProfile `json:"old,omitempty"`
+	// intermediate is a security profile based on:
+	//
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28default.29
+	//
+	// and looks like this (yaml):
+	//
+	//   ciphers:
+	//     - ECDHE-ECDSA-CHACHA20-POLY1305
+	//     - ECDHE-RSA-CHACHA20-POLY1305
+	//     - ECDHE-ECDSA-AES128-GCM-SHA256
+	//     - ECDHE-RSA-AES128-GCM-SHA256
+	//     - ECDHE-ECDSA-AES256-GCM-SHA384
+	//     - ECDHE-RSA-AES256-GCM-SHA384
+	//     - DHE-RSA-AES128-GCM-SHA256
+	//     - DHE-RSA-AES256-GCM-SHA384
+	//     - ECDHE-ECDSA-AES128-SHA256
+	//     - ECDHE-RSA-AES128-SHA256
+	//     - ECDHE-ECDSA-AES128-SHA
+	//     - ECDHE-RSA-AES256-SHA384
+	//     - ECDHE-RSA-AES128-SHA
+	//     - ECDHE-ECDSA-AES256-SHA384
+	//     - ECDHE-ECDSA-AES256-SHA
+	//     - ECDHE-RSA-AES256-SHA
+	//     - DHE-RSA-AES128-SHA256
+	//     - DHE-RSA-AES128-SHA
+	//     - DHE-RSA-AES256-SHA256
+	//     - DHE-RSA-AES256-SHA
+	//     - ECDHE-ECDSA-DES-CBC3-SHA
+	//     - ECDHE-RSA-DES-CBC3-SHA
+	//     - EDH-RSA-DES-CBC3-SHA
+	//     - AES128-GCM-SHA256
+	//     - AES256-GCM-SHA384
+	//     - AES128-SHA256
+	//     - AES256-SHA256
+	//     - AES128-SHA
+	//     - AES256-SHA
+	//     - DES-CBC3-SHA
+	//     - !DSS
+	//   securityProtocol:
+	//     minimumVersion: TLSv1.0
+	//     maximumVersion: TLSv1.2
+	//   dhParamSize: 2048
+	//
+	// +optional
+	// +nullable
+	Intermediate *IntermediateSecurityProfile `json:"intermediate,omitempty"`
+	// modern is a security profile based on:
+	//
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Modern_compatibility
+	//
+	// and looks like this (yaml):
+	//
+	//   ciphers:
+	//     - ECDHE-ECDSA-AES256-GCM-SHA384
+	//     - ECDHE-RSA-AES256-GCM-SHA384
+	//     - ECDHE-ECDSA-CHACHA20-POLY1305
+	//     - ECDHE-RSA-CHACHA20-POLY1305
+	//     - ECDHE-ECDSA-AES128-GCM-SHA256
+	//     - ECDHE-RSA-AES128-GCM-SHA256
+	//     - ECDHE-ECDSA-AES256-SHA384
+	//     - ECDHE-RSA-AES256-SHA384
+	//     - ECDHE-ECDSA-AES128-SHA256
+	//     - ECDHE-RSA-AES128-SHA256
+	//   securityProtocol:
+	//     minimumVersion: TLSv1.2
+	//     maximumVersion: TLSv1.2
+	//   dhParamSize: 2048
+	//
+	// +optional
+	// +nullable
+	Modern *ModernSecurityProfile `json:"modern,omitempty"`
+	// custom is a user-defined security profile. An example custom profile
+	// looks like this:
+	//
+	//   ciphers:
+	//     - ECDHE-ECDSA-CHACHA20-POLY1305
+	//     - ECDHE-RSA-CHACHA20-POLY1305
+	//     - ECDHE-RSA-AES128-GCM-SHA256
+	//     - ECDHE-ECDSA-AES128-GCM-SHA256
+	//   securityProtocol:
+	//     minimumVersion: TLSv1.1
+	//     maximumVersion: TLSv1.2
+	//   dhParamSize: 1024
+	//
+	// Any unset fields of a custom security profile default to the IntermediateSecurityProfile.
```

Comment:
Not sure that "IntermediateSecurityProfile" is a meaningful reference... maybe refer to the Type value instead, like

>Any unset fields default to the equivalent field of the Intermediate profile type.